### PR TITLE
Bugfix/handle non existent current user

### PR DESCRIPTION
**Pull request by JoeAtHPI**
This PR adds error handling when an unsigned user tries to access the website. This changes how one should handle the tests - by adding tags seen in the `lectures_controller_spec.rb` one can add who is logged in right now

### CODE REVIEW

File: spec/controllers/lectures_controller_spec.rb [hunk 1]
```diff
@@ -104,13 +123,12 @@
...
+      it "redirects for other lecturers", :logged_lecturer do
         lecture = Lecture.create! valid_attributes_with_lecturer
-        login_lecturer()
         put :update, params: { id: lecture.to_param, lecture: invalid_attributes }, session: valid_session
         expect(response).to redirect_to(lectures_url)
       end
-      it "returns a success response for lecturer (i.e. to display the 'edit' template)" do
+      it "returns a success response for lecturer (i.e. to display the 'edit' template)", :logged_lecturer do
         lecture = Lecture.create! valid_attributes_with_lecturer
         login_lecturer(lecture.lecturer)
```
Comment (MichaelBuessemeyer) [inline spec/controllers/lectures_controller_spec.rb]: I am not sure, but is this still needed? The tag `:logged_lecturer` should already log in an lecturer. Here it looks / reads like we are logging the lecturer in again 🤔

Comment (MichaelBuessemeyer) [reply spec/controllers/lectures_controller_spec.rb]: I mean the `login_lecturer(lecture.lecturer)` in line 115/133

**Comment by MichaelBuessemeyer** [comment]
And next time: please assign yourself and add appropriate tags :)

File: spec/controllers/lectures_controller_spec.rb (hunk 1 above)
Comment (grittaweisheit) [reply spec/controllers/lectures_controller_spec.rb]: Yes it is because it's a special lecturer

**Pull request merged**